src/controllers/open-data/open-data.js

```diff
@@ -0,0 +1,154 @@
+  let id = data[0] // The id of the dataset
+  let dataset = data[1] // The name of the dataset
+  let column = data[2]
+  let key = `opendata/column/${id}/${column}` // Unique key for dataset storage
+  let url = `https://data.edmonton.ca/resource/${id}.json` // The Socrata api endpoint
+  let queryBase = `${url}?$query=`
+
+  let storedData = await req.cache.getLatest(key)
+  let storedTimestamp
+  let filteredStoredColumnValues // The stored/previous column values to be compared to
+  if (storedData) {
+    storedTimestamp = storedData.created_at
+    filteredStoredColumnValues = JSON.parse(storedData.all_values).map(row => {
+      return row[column]
+    })
+  } else {
+    storedTimestamp = '1998-07-25T00:00:00.000Z' // This value is random
+  }
+
+  // TODO: Change to appropriate limit
+  // Get all the columns (filter it by column later)
+  let getLatestUpdatedQuery = `${queryBase}SELECT * WHERE :updated_at >= '${storedTimestamp}' ORDER BY :updated_at DESC LIMIT 100`
+  let getUpdatedTimestamp = `${queryBase}SELECT :updated_at ORDER BY :updated_at LIMIT 1`
+  let latestTimestamp
+  let latestColumnRows
+  let filteredLatestColumnValues // The current values of the column being monitored
+  try {
+    let rawJsonUpdatedAt = await request(getUpdatedTimestamp)
+    latestTimestamp = JSON.parse(rawJsonUpdatedAt)[0][':updated_at']
+    let rawJsonColumnValues = await request(getLatestUpdatedQuery)
+    latestColumnRows = JSON.parse(rawJsonColumnValues)
+    filteredLatestColumnValues = latestColumnRows.map(row => {
+      return row[column]
+    })
+  } catch (e) {
+    e.code = 500
+    throw e
+  }
+  // Ensure that the date is in ISO 8601 time format
+  // TODO: Check for other potential time formats
+  if (
+    !latestTimestamp
+      .toString()
+      .match(/[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}Z/)
+  ) {
+    // Convert Epoch times to ISO 8601 time format
+    latestTimestamp = new Date(latestTimestamp * 1000).toISOString()
+  }
+
+  let responseValues
+  if (storedData && storedTimestamp == latestTimestamp) {
+    console.log('Dataset rows not updated. Returning old data.')
+    responseValues = await req.cache.getAll(key, req.body['limit'])
+  } else {
+    // Dataset rows were updated - now check if the columns are different
+    console.log('Dataset rows updated.')
+    if (
+      storedData &&
+      compareArr(filteredStoredColumnValues, filteredLatestColumnValues)
+    ) {
+      console.log('Row values not changed. Returning old data')
+      responseValues = await req.cache.getAll(key, req.body['limit'])
+    } else {
+      console.log('Adding new rows')
+      let id = uuid()
+      // The unique difference (does not catch repeat values)
+      let diff = {
+        new: arrDiff(filteredLatestColumnValues, filteredStoredColumnValues),
+        removed: arrDiff(filteredStoredColumnValues, filteredLatestColumnValues)
+      }
+      let newRows = {
+        id,
+        created_at: latestTimestamp,
```

Comment:
Avoid using snake case if possible. The reason I have some snake case in my code is because I need to match the style of the old PHP. Changing the ingredients you return after going to production is a pain.

Reply:
I think it has to match the IFTTT slug (which is in snake case)